site/.build.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+const module_path = './ts/';
+
+let files = [];
+
+function getAllFiles(path){
+    fs.readdirSync(path, {withFileTypes:true}).forEach((file) => {
+        if (file.isDirectory()){
+            return getAllFiles(`${path}${file['name']}/`);
+        }
+
+        const filename = file['name'];
+        if (!filename.includes('.js') && !filename.includes('.ts')){
+            return;
+        }
+
+        files.push(path + filename);
+    });
+}
+
+getAllFiles(module_path);
```

Comment:
Let's rewrite this as:

```suggestion
function getAllFiles(dir) {
    return fs.readdirSync(dir, { withFileTypes: true }).reduce((acc, entry) => {
        if (!entry.isDirectory() && entry.name.endsWith('.ts') || entry.name.endsWith('.js')) {
            acc.push(path.join(dir, entry.name));
        }
        
        return acc;
    }, []);
}

const files = getAllFiles(module_path);
```

Better to have a pure function than one that has side-effects. Let's also not introduce recursion until we need it, especially with regards to handling `entryPoints` below.